sim/core/dot.go

```diff
@@ -230,6 +230,9 @@ func NewDot(config Dot) *Dot {
 type DotArray []*Dot
 
 func (dots DotArray) Get(target *Unit) *Dot {
+	if len(dots) < 1 {
```

Comment:
This function gets called a lot, so I don't want to add a check here for performance reasons. Where is an issue occurring that this was needed?

Reply:
Removing the part I added will cause a runtime error to occur when I check for Garrote Dot to determine if I should be casting it. I check the dot by using `rogue.Garrote.Dot(rogue.CurrentTarget)` and the same Dot call for rupture.

Stack trace for running without Dot fix above is this:
`Simulation Failure:
runtime error: index out of range [0] with length 0
Stack Trace:
goroutine 71 [running]:
runtime/debug.Stack()
	/usr/lib/go/src/runtime/debug/stack.go:24 +0x65
github.com/wowsims/wotlk/sim/core.runSim.func1()
	/home/catszeid/Programs/wotlk/sim/core/sim.go:59 +0x9e
panic({0xcca580, 0xc0003441f8})
	/usr/lib/go/src/runtime/panic.go:838 +0x207
github.com/wowsims/wotlk/sim/core.DotArray.Get(...)
	/home/catszeid/Programs/wotlk/sim/core/dot.go:233
github.com/wowsims/wotlk/sim/core.(*Spell).Dot(...)
	/home/catszeid/Programs/wotlk/sim/core/spell.go:279
github.com/wowsims/wotlk/sim/rogue.(*Rogue).setSubtletyBuilder(0x7f5e30578470?)
	/home/catszeid/Programs/wotlk/sim/rogue/subtlety_rotation.go:19 +0x15b
github.com/wowsims/wotlk/sim/rogue.(*Rogue).setPriorityItems(0xc000604000, 0xc0006722c0)
	/home/catszeid/Programs/wotlk/sim/rogue/rotation.go:271 +0xe5
github.com/wowsims/wotlk/sim/rogue.(*Rogue).Reset(0xc000604000, 0xc0006722c0)
	/home/catszeid/Programs/wotlk/sim/rogue/rogue.go:228 +0x105
github.com/wowsims/wotlk/sim/core.(*Character).reset(0xc000604000, 0x200?, {0x2e86eb8, 0xc000604000})
	/home/catszeid/Programs/wotlk/sim/core/character.go:437 +0xcc
github.com/wowsims/wotlk/sim/core.(*Party).reset(0xc00012cea0, 0xc00076bd30?)
	/home/catszeid/Programs/wotlk/sim/core/raid.go:75 +0x98
github.com/wowsims/wotlk/sim/core.(*Raid).reset(0xc000146280, 0xd7b393?)
	/home/catszeid/Programs/wotlk/sim/core/raid.go:417 +0xd4
github.com/wowsims/wotlk/sim/core.(*Simulation).reset(0xc0006722c0)
	/home/catszeid/Programs/wotlk/sim/core/sim.go:237 +0x271
github.com/wowsims/wotlk/sim/core.(*Simulation).runOnce(0xc0006722c0)
	/home/catszeid/Programs/wotlk/sim/core/sim.go:323 +0x25
github.com/wowsims/wotlk/sim/core.(*Simulation).run(0xc0006722c0)
	/home/catszeid/Programs/wotlk/sim/core/sim.go:273 +0x177
github.com/wowsims/wotlk/sim/core.runSim(0xfffffffc3bfe3b00?, 0xc000322d80, 0x0)
	/home/catszeid/Programs/wotlk/sim/core/sim.go:113 +0x2a6
github.com/wowsims/wotlk/sim/core.RunSim(0xc00004ef94?, 0x2d467e8?)
	/home/catszeid/Programs/wotlk/sim/core/sim.go:45 +0x1b
created by github.com/wowsims/wotlk/sim/core.RunRaidSimAsync
	/home/catszeid/Programs/wotlk/sim/core/api.go:45 +0x72`

Reply:
Fixed with merging master in again